### PR TITLE
order commit to project bookshop

### PR DESCRIPTION
**Pull request by OlexiiMelnik**

### CODE REVIEW

File: src/main/java/online/bookshop/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return orderService.create(user.getId(), requestDto);
+    }
+
+    @Operation(summary = "User order history",
+            description = "Find all the orders by user")
+    @GetMapping
+    public List<OrderResponseDto> findAllByUserId(
+            Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.findAllByUserId(user.getId(), pageable);
```
Comment (aaananas) [inline src/main/java/online/bookshop/controller/OrderController.java:51]:
```suggestion
    @Operation(summary = "User order history",
            description = "Find all the orders by user")
    @GetMapping
    public List<OrderResponseDto> findAllByCurrentUser(
            Authentication authentication, Pageable pageable) {
        User user = (User) authentication.getPrincipal();
        return orderService.findAllByUserId(user.getId(), pageable);
```

File: src/main/java/online/bookshop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    public List<OrderResponseDto> findAllByUserId(Long userId, Pageable pageable) {
+        List<Order> allByUserId = orderRepository.findAllByUserId(userId);
+        return allByUserId.stream()
+                .map(order -> {
+                    OrderResponseDto orderResponseDto = orderMapper.toDto(order);
+                    Set<OrderItemResponseDto> orderItemResponseDtos
+                            = changeToItemResponseDto(order.getOrderItems());
+                    orderResponseDto.setOrderItems(orderItemResponseDtos);
+                    return orderResponseDto;
+                })
```
Comment (aaananas) [inline src/main/java/online/bookshop/service/impl/OrderServiceImpl.java]: make a mapper for this

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/online/bookshop/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (OlexiiMelnik) [reply src/main/java/online/bookshop/service/impl/OrderServiceImpl.java]: i do in changeToItemResponseDto

File: src/main/java/online/bookshop/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    public OrderResponseDto updateStatus(@PathVariable Long id,
+                                         @RequestBody OrderUpdateStatusDto orderUpdateStatusDto) {
+        return orderService.updateStatus(id, orderUpdateStatusDto);
+    }
+
+    @Operation(summary = "Find all orderItems by order",
+            description = "Find all orderItems by order")
+    @GetMapping("/{id}/items")
+    public Set<OrderItemResponseDto> findAllOrderItems(@PathVariable Long id) {
+        return orderService.findAllOrderItems(id);
```
Comment (Rommelua) [inline src/main/java/online/bookshop/controller/OrderController.java:67]: anyone can see any Order by id? Use Authentication

File: src/main/java/online/bookshop/controller/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Operation(summary = "Find orderItem by orderId and itemId",
+            description = "Find orderItem by orderId and itemId")
+    @GetMapping("/{orderId}/items/{itemId}")
+    public OrderItemResponseDto findOrderItemById(
+            @PathVariable Long orderId, @PathVariable Long itemId) {
+        return orderService.findOrderItemById(orderId, itemId);
```
Comment (Rommelua) [inline src/main/java/online/bookshop/controller/OrderController.java:75]: same

File: src/main/java/online/bookshop/model/OrderItem.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        OrderItem orderItem = (OrderItem) o;
+        return Objects.equals(quantity, orderItem.quantity)
+                && Objects.equals(price, orderItem.price);
```
Comment (Rommelua) [inline src/main/java/online/bookshop/model/OrderItem.java]: add Book

File: src/main/java/online/bookshop/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                getOrderItemsFromShoppingCart(shoppingCart, order);
+        order.setOrderItems(orderItems);
+        BigDecimal total = calculateTotal(orderItems);
+        order.setTotal(total);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        OrderResponseDto orderResponseDto
+                = orderMapper.toDto(order);
+        orderResponseDto.setOrderItems(changeToItemResponseDto(orderItems));
+        return orderResponseDto;
```
Comment (Rommelua) [inline src/main/java/online/bookshop/service/impl/OrderServiceImpl.java]: ShoppingCart should be empty now.

File: src/main/java/online/bookshop/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/online/bookshop/service/impl/OrderServiceImpl.java]: mapper should do it 
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/online/bookshop/controller/OrderController.java (hunk 3 above)
Comment (OlexiiMelnik) [reply src/main/java/online/bookshop/controller/OrderController.java:67]: GET: /api/orders/{orderId}/items
GET: /api/orders/{orderId}/items/{itemId} it is hw

File: src/main/java/online/bookshop/controller/OrderController.java (hunk 4 above)
Comment (OlexiiMelnik) [reply src/main/java/online/bookshop/controller/OrderController.java:75]: GET: /api/orders/{orderId}/items
GET: /api/orders/{orderId}/items/{itemId} it is hw

**Review by aaananas — APPROVED** [review]

**Pull request merged**